src/BackgroundProcess/Application/Port/Out/AsyncRequest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Jetty\BackgroundProcessing\BackgroundProcess\Application\Port\Out;
+
+use Jetty\BackgroundProcessing\BackgroundProcess\Exception\AsyncException;
+
+/**
+ * Describes the minimum functionality needed for an async request.
+ */
+interface AsyncRequest
+{
+    /**
+     * Sets the data that is sent and received on both sides of the request.
+     *
+     * @param array $data Data.
+     */
+    public function data(array $data): AsyncRequest;
+
+
+    /**
+     * Dispatch the async request.
+     *
+     * @return array<string, string>
+     *
+     * @throws AsyncException
+     */
+    public function dispatch(): array;
```

Comment:
Do we care what is returned from this method? I wouldn't think so: it either dispatched or didn't (exception). And, since it is asynchronous, we won't be getting anything meaningful back, because the job hasn't finished running yet.

Reply:
Like it or not, this is the existing interface for the function. This will likely get reworked in the future.